File: Code/Arrays_Bit_Manipulation/Best_Time_To_Buy_And_Sell_Stock_I.cs
using System;

class Program {
    static int MaxProfit(int[] prices) {
        int minPrice = int.MaxValue;
        int maxProfit = 0;

        foreach (int price in prices) {
            if (price < minPrice) {
                minPrice = price;
            } else if (price - minPrice > maxProfit) {
                maxProfit = price - minPrice;
            }
        }

        return maxProfit;
    }

    static void Main() {
        Console.Write("Enter number of days: ");
        int n = int.Parse(Console.ReadLine());

        int[] prices = new int[n];
        Console.WriteLine("Enter stock prices:");
        string[] inputs = Console.ReadLine().Split(' ');

        for (int i = 0; i < n; i++) {
            prices[i] = int.Parse(inputs[i]);
        }

        int result = MaxProfit(prices);
        Console.WriteLine("Maximum profit: " + result);
    }
}

File: Code/Arrays_Bit_Manipulation/Best_Time_To_Buy_And_Sell_Stock_I.py
def maxProfit(prices):
    min_price = float('inf')
    max_profit = 0

    for price in prices:
        if price < min_price:
            min_price = price
        elif price - min_price > max_profit:
            max_profit = price - min_price

    return max_profit

if __name__ == "__main__":
    n = int(input("Enter number of days: "))
    prices = list(map(int, input("Enter stock prices: ").split()))
    result = maxProfit(prices)
    print("Maximum profit:", result)
